### ISSUE TITLE
Documentation of the file headers in the catlas output 

### ISSUE THREAD

**Issue by drtamermansour** [spacegraphcats/spacegraphcats#440]
The catlas output files are described at a couple places. I found [this page](https://spacegraphcats.github.io/spacegraphcats/01-running-spacegraphcats/) the most comprehensive. However, the descriptions are always missing clear definitions of the file headers. I am starting this issue to document these files

**Comment by drtamermansour** [comment]
```results.csv```  

- Where: in the output directory of a search. 
- File description: summary results for the queries (containment, similarity, etc.)
- Headers: query,containment,similarity,bp,contigs,ksize,num_query_kmers,best_containment,cdbg_min_overhead,catlas_min_overhead,catlas_name
- Description of the headers (My understanding. It could be wrong)
    - query: The query file with an absolute path
    - containment: how much of the query is contained in the Catlas (value: 0-1)
    - similarity: jaccard distance between the query and the matching Catlas compartments (value: 0-1)
    - bp: The numbers of base pairs in the matching Catlas compartments
    - contigs: The numbers of cbdg unitigs in the matching Catlas compartments
    - ksize: The kmer size used in the building the Catlas
    - num_query_kmers: The numbers of kmers in this query file
    - best_containment: If the query is matching multiple Catlas compartments, containment will be calculated for each compartment. This column reports the highest containment (value: 0-1)
    - cdbg_min_overhead: ??
    - catlas_min_overhead:?? The numbers of kmers in the matching Catlas compartments that does not match query kmers
    - catlas_name: The name of the Catlas used in this search

**Comment by drtamermansour** [comment]
``` <query_file>.response.txt ```

- Where: in the output directory of a search.
- File description: response curve showing how much overhead is gained for each node
- Headers:    sum_cont relative_cont relative_overhead sum_cont2 sum_oh catlas_id
- Description of the headers (Again this is my guessing)
    - sum_cont: cumulative kmer count of the query sequences
    - relative_cont: cumulative kmer fraction of the query sequences
    - relative_overhead: The (?cumulative) numbers of kmers in the matching Catlas compartments that does not match query kmers
    - sum_cont2: kmer count of the query sequence (this is a non-cumulative value)
    - sum_oh: ?non-cumulative overhead
    - catlas_id: ?the id(s) of the matching Catlas compartments

**Comment by ctb** [comment]
> `results.csv`
> 
>     * Where: in the output directory of a search.
> 
>     * File description: summary results for the queries (containment, similarity, etc.)

Yes. These are all calculated in `spacegraphcats/search/query_by_sequence.py`.
 
>     * Headers: query,containment,similarity,bp,contigs,ksize,num_query_kmers,best_containment,cdbg_min_overhead,catlas_min_overhead,catlas_name
> 
>     * Description of the headers (My understanding. It could be wrong)
>       
>       * query: The query file with an absolute path

✅ 

>       * containment: how much of the query is contained in the Catlas (value: 0-1)

✅  - estimated by sourmash using signatures calculated from the query and those contigs retrieved by the neighborhood query.

>       * similarity: jaccard distance between the query and the matching Catlas compartments (value: 0-1)

✅  - estimated by sourmash using signatures calculated from the query and those contigs retrieved by the neighborhood query.

>       * bp: The numbers of base pairs in the matching Catlas compartments

✅   - Yep, these are the total bp in contigs retrieved by the neighborhood query (literally, sum of length of contigs).

>       * contigs: The numbers of cbdg unitigs in the matching Catlas compartments

✅   - total number of contigs retrieved by the neighborhood query (literally, sum of length of contigs).

>       * ksize: The kmer size used in the building the Catlas

✅

>       * num_query_kmers: The numbers of kmers in this query file

✅ - the number of _distinct_ k-mers in the query file

>       * best_containment: If the query is matching multiple Catlas compartments, containment will be calculated for each compartment. This column reports the highest containment (value: 0-1)

Nope! This turns out to be debugging/tracking leftover from earlier efforts.

Technically, it is the total number of actual matching k-mers in the entire catlas, divided by the size of the query.

Unlike `containment` above, it is not an estimate (based on sourmash), but should be exact. But, on the flip side, it is calculated from summary information annotated on the catlas, and not from actual sequence.

If this number is different from `containment`, it means that Something Has Gone Wrong Somewhere.

(In pre-1.0 work, when we had different ways of finding things in the catlas, there were other meanings for this number.)

>       * cdbg_min_overhead: ??

This is the number of 'excess' k-mers (above and beyond the query k-mers) that have been retrieved in matching cDBG nodes, without using the dominators; normalized by total found cdbg node size.

The equation is 'sum(size(cdbg_matching_nodes)) - size(query_found_kmers) / sum(size(cdbg_matching_nodes))`.

If this is zero, then the cDBG nodes that were found were exact matches to the query, and no excess k-mers were retrieved.

This cannot be exactly 1, because that would require that size(query_found_kmers) be 0, that is, there are no k-mers matching between the cDBG and the query.

>       * catlas_min_overhead:?? The numbers of kmers in the matching Catlas compartments that does not match query kmers

This is the number of `excess` k-mers that have been retrieved in matching neighborhoods.

The equation is `sum(size(nodes_in_matching_neighborhoods)) - size(query_found_kmers) / sum(size(nodes_in_matching_neighborhoods))`.

If this is zero, then the neighborhoods  that were found were exact matches to the query, and no excess k-mers were retrieved. That is, nothing new was found in the catlas.

This cannot be exactly 1, because that would require that size(query_found_kmers) be 0, that is, there are no k-mers matching between the catlas and the query.

>       * catlas_name: The name of the Catlas used in this search

✅ - right, this is the directory name, so something like `dory_k21_r1`.
